src/hooks/useScrollFadeStyles.mdx

```diff
@@ -0,0 +1,60 @@
+import { Meta, Source } from "@storybook/blocks";
+
+<Meta title="Hooks/useScrollFadeStyles" />
+
+# useScrollFadeStyles
+
+| Hook Version | DS Version |
+| ------------ | ---------- |
+| Added        | `Prerelease`   |
+| Latest       | `Prerelease`   |
+
+
+ The `useScrollFadeStyles` hook provides a way to manage the visibility of a fade effect on the right edge of a horizontally scrollable element.
+ It monitors the scroll position and determines whether the fade effect should be shown or hidden based on the scroll state.
+
+## Usage
+
+The hook useScrollFadeStyles returns an object with both the scrollableRef and showRightFade boolean value.
+
+<Source
+  code={`
+const {
+  scrollableRef,
+  showRightFade,
+} = useScrollFadeStyles();
+`}
+  language="jsx"
+/>
+
+<Source
+  code={`
+// ...
+<Box>
+  <div>
+    <List
+      type="ul"
+      m="0"
+      ref={scrollableRef}
+      inline
+      noStyling
+    >
+      <>
+        <li>
+          <SubNavButton id="subnav-action">
+            Action
+          </SubNavButton>
+        </li>
+        <li>
+          <SubNavLink id="subnav-link">
+            Link
+          </SubNavLink>
+        </li>
+      </>
+    </List>
+    {showRightFade && <Box sx={styles.fadeEffect} />}
```

Comment:
@deepikagonuguntla Can you also include a code example showing the styles?

Reply:
Ok